x-pack/plugins/ingest_manager/server/services/agents/enroll.ts

```diff
@@ -18,6 +20,12 @@ export async function enroll(
   metadata?: { local: any; userProvided: any },
   sharedId?: string
 ): Promise<Agent> {
+  const kibanaVersion = appContextService.getKibanaVersion();
+  const version: string | undefined = metadata?.local?.elastic?.agent?.version;
+  if (!version || semver.compare(version, kibanaVersion) === 1) {
```

Comment:
I think `.lte` or another comparison function is easier to understand. I had to look up the compare and it results to understand what was happening.

Do we need to set the includePrerelease option https://github.com/npm/node-semver#functions to deal with values like `8.0.0-SNAPSHOT`?